usaspending_api/download/filestreaming/csv_generation.py

```diff
@@ -55,9 +55,9 @@ def generate_csvs(download_job, sqs_message=None):
         download_job.file_size = os.stat(file_path).st_size
     except Exception as e:
         # Set error message; job_status_id will be set in generate_zip.handle()
-        download_job.error_message = 'An exception was raised while attempting to write the file:\n{}'.format(e)
+        download_job.error_message = 'An exception was raised while attempting to write the file:\n{}'.format(str(e))
         download_job.save()
-        raise Exception(download_job.error_message)
+        raise type(e)(download_job.error_message)
```

Comment:
This is pretty cool. So this will re-raise the exception but with the specific sub-type?

Reply:
Yep. When I added an `InvalidParameterException` to `csv_source.py`, it was then being cast as a regular `Exception` here. So this will make sure that any exception raised below it will be the right type.